joblib/parallel.py

```diff
@@ -211,6 +213,24 @@ def _get_active_backend(
         thread_config['n_jobs'] = 1
         return sharedmem_backend, thread_config
 
+    force_processes = (
+        not explicit_backend and prefer == 'processes' and uses_threads
+    )
+
+    if force_processes:
+        # This backend does not match the prefer="processes" constraint:
+        # fallback to the default process-based backend.
+        process_backend = BACKENDS[DEFAULT_PROCESS_BACKEND](
+            nesting_level=nesting_level
+        )
+        # Force to n_jobs=1 by default
+        process_config = backend_config.copy()
+        # TODO Uncommenting the next line breaks the tests, I have to say I
+        # don't understand this why is it needed in the process => threads
+        # forcing above
+        # process_config['n_jobs'] = 1
```

Comment:
I need this commented out otherwise one of the test fails, I don't understand why it is done like this for switching to threads above and why this is needed here ...

Reply:
I don't even understand why we use `thread_config['n_jobs'] = 1` in the `if force_threads` branch of the code above.

Reply:
Let's remove both lines?